grip_api/src/grip_api/task_editor_widgets/terminal_socket.py

```diff
@@ -33,102 +33,72 @@ def __init__(self, container, socket_name, index, multi_connections=True):
             @param index: Index of the socket
             @param multi_connections: Indicate whether the socket can host several connectors. Default to True
         """
-        # Set the id of the object
-        self.id = id(self)
+        super().__init__(socket_name, index, multi_connections)
         # Store the Container the socket is added to
         self.container = container
-        # Name of the outcome corresponding to the socket
         self.name = socket_name
         # Store whether this terminal socket is the one marking the beginning of the state machine or not
         self.is_starting = self.name == "Start"
-        self.index = index
-        # Set the multi connected attribute
-        self.is_multi_connected = multi_connections
-        # Create and store the graphical socket to be displayed
-        self.graphics_socket = TerminalGraphicsSocket(self)
-        # Get all the connectors connected to this terminal socket
-        self.connectors = list()
-        # Allows to recognize terminal sockets from others when releasing edges
-        self.is_terminal = True
         # By default a terminal socket can't be deleted from the container
         self.is_deletable = False
```

Comment:
Maybe this should be an initilisation argument

Reply:
I've also changed call to TerminalSocket in container.py to include `is_deletable=True`